main.js

```diff
@@ -0,0 +1,142 @@
+// The app collection
+const AppCollection = Backbone.Collection.extend({
+  // grabs the url thats passed down from the input
+  url() {
+    return this.models[0].attributes.url;
```

Comment:
Never access `attributes` directly.

We should not be defining urls on these models (we can't even call them `videoModels` necessarily because you have not defined that anywhere.